ebean-core/src/test/java/org/tests/idkeys/TestGeneratedKeys.java

```diff
@@ -49,30 +48,43 @@ public void testSequence() throws SQLException {
   }
 
   private long readSequenceValue(Transaction tx, String sequence) throws SQLException {
-    Statement stm = null;
-    try {
-      stm = tx.getConnection().createStatement();
-      ResultSet rs = stm.executeQuery("select currval('" + sequence + "')");
-      rs.next();
-
-      return rs.getLong(1);
-    } finally {
-      if (stm != null) {
-        try {
-          stm.close();
-        } catch (SQLException e) {
-        }
+    try (Statement stm = tx.getConnection().createStatement()) {
+      ResultSet rs;
+
+      switch (spiEbeanServer().getDatabasePlatform().getPlatform()) {
+        case H2 :
+          rs = stm.executeQuery("select currval('" + sequence + "')");
+          rs.next();
+          return rs.getLong(1);
+
+        case DB2 :
+          rs = stm.executeQuery("values previous value for " + sequence);
+          rs.next();
+          return rs.getLong(1);
+
+        case MARIADB :
+          rs = stm.executeQuery("select previous value for " + sequence);
+          rs.next();
+          return rs.getLong(1);
+
+        case SQLSERVER :
+          rs = stm.executeQuery(
```

Comment:
We can get some reuse here.

Reply:
I've used the 🧹 to tidy up the code. MariaDB seems not to support this approach.